### PR TITLE
twtxt improvements

### PR DESCRIPTION
**Pull request by JonLundy**
##### Summary

feat: improve json output with mentions/links

This adds the links and mention text to the json object. I just have it as the literal text.. maybe we can expand them split as nick/url for mention and text/url for links?

feat: change twter to pointer so can be updated by feed nick/url values

change internal twter for a twtfile to be a pointer so changing the file twter changes all twters

### CODE REVIEW

File: types/lextwt/lextwt.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ func ParseFile(r io.Reader, twter types.Twter) (types.TwtFile, error) {
 	}
 
 	if v, ok := f.Info().GetN("nick", 0); ok {
+		log.Warnf("override nick %s with %s", f.twter.Nick, v.Value())
```
Comment (der-lyse) [inline types/lextwt/lextwt.go:48]: I didn't run the code, but it looks like this warning is always issued when encountering a nick comment, even though, the nick might not have changed.

**Review by der-lyse — COMMENTED** [review]
Cool!

File: types/lextwt/lextwt.go (hunk 1 above)
Comment (JonLundy) [reply types/lextwt/lextwt.go:48]: true.. probably can drop it down to debug

**Comment by prologic** [comment]
Is this ready to merge?

**Comment by JonLundy** [comment]
oh.. yeah i think so. i had some other things i was working on(fixing mention expands) but i can do that on a new PR.

**Comment by JonLundy** [comment]
@prologic good to go now!

**Pull request merged**